syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.iotservice";
option java_outer_classname = "IoTServiceProto";
option objc_class_prefix = "TEMP";

package DBList_service;

service DBListService {
  
  // Retorna a lista
  rpc GetList (Null) returns (List) {}
  
  // Retorna o valor do index
  rpc GetValue (Index) returns (Value) {}
  
  // Adiciona um valor a lista
  rpc Append (Value) returns (Null) {}
  
  // Retora o indice do primeiro elemento com valor Value
  rpc SearchValue (Value) returns (Index) {}

  // Remove o elemento de index Index da lista
  rpc RemoveIdx (Index) returns (Null) {}

  //Remove o elemento com valor Value
  rpc RemoveValue (Value) returns (Null) {}

  //Ordena os elementos da lista
  rpc Sort (Null) returns (Null) {}

  //Reverte os elementos da lista
  rpc Reverse (Null) returns (Null) {}
  
  //Retorna o tamanhao da lista
  rpc GetLength (Null) returns (Size) {}
  
}

message Index {
  uint32 id = 1;
}

message Value {
  uint32 value = 1;
}

message Size {
    uint32 size = 1;
}

message List {
  repeated Value values = 1;
}

message Null {}